src/persistenttabmodel.cpp

```diff
@@ -36,10 +36,12 @@ void PersistentTabModel::tabsAvailable(QList<Tab> tabs)
 {
     beginResetModel();
     int oldCount = count();
-    m_tabs.clear();
-    m_tabs = tabs;
 
```

Comment:
Actually... would following do the same trick. I think there was and is possible error, at least code is error phone. 

```
// Clear always previous tabs
clear();
if (m_tabs.count() > 0) {
    m_tabs = tabs;
...
```
```